ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/common/HttpRequest.kt

```diff
@@ -0,0 +1,83 @@
+package com.izivia.ocpi.toolkit.common
+
+import com.izivia.ocpi.toolkit.common.context.currentResponseMessageRoutingHeadersOrNull
+import com.izivia.ocpi.toolkit.transport.domain.HttpRequest
+import com.izivia.ocpi.toolkit.transport.domain.HttpResponse
+import com.izivia.ocpi.toolkit.transport.domain.HttpStatus
+import org.apache.logging.log4j.LogManager
+import java.time.Instant
+
+private val logger = LogManager.getLogger(HttpRequest::class.java)
+
+suspend fun <T> HttpRequest.respondSearchResult(now: Instant, fn: suspend () -> SearchResult<T>) = wrapResponse {
```

Comment:
return value specific helper for improved readability, avoids some ifs and type checks